pkg/controller/direct/logging/utils.go

```diff
@@ -317,15 +306,9 @@ func convertKCCtoAPIForLogMetricLabels(kccLabels []krm.LogmetricLabels) []*api.L
 	for i, kccLabel := range kccLabels {
 		apiLabels[i] = &api.LabelDescriptor{}
 
-		if kccLabel.Description != nil {
-			apiLabels[i].Description = *kccLabel.Description
-		}
-		if kccLabel.Key != nil {
-			apiLabels[i].Key = *kccLabel.Key
-		}
-		if kccLabel.ValueType != nil {
-			apiLabels[i].ValueType = *kccLabel.ValueType
-		}
```

Comment:
are we losing the "unset" intent if we are doing this? What was previously "nothing"/ unset would now be ""

Reply:
(Sorry I missed your comment before approve/lgtm)
I think this is intended: the KCC object should be declarative meaning if it is "unset", users expect the server-side value to be nil/unset.  

One potential problem though is the `spec.Filter` field. It is a required field and when updating on the server side, the server will ignore the "unset" `filter` but keeping the orignal value. For that case, we shall make `spec.filter` required field. (I think I raised this question in some other chats).

Reply:
For the above potential problem, I think we are good as well (checked the CRD) https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/4c4bccd8d7390a4d217e451fbbe6ac5936d80fcb/crds/logging_v1beta1_logginglogmetric.yaml#L346